doc/nrf/ug_dev_model.rst

```diff
@@ -0,0 +1,514 @@
+want to obtain a copy of the |NCS| on your filesystem and then manage the
+source code of both the SDK and your application yourself.
+
+Since no downloadable packages of the |NCS| are currently made available, the
+simples path to obtain the source code is to follow the instructions in the
+:ref:`corresponding section <dm-wf-get-ncs>` of the documentation. This will
+require you to install Git and west, but you can then ignore them from that
+point onwards. If you need to update the copy of the |NCS| you are working with
+you can :ref:`obtain the source code <dm-wf-get-ncs>` again or, if you have
+kept the original set of repositories,
+:ref:`update it instead <dm-wf-update-ncs>`.
+Once you have obtained a copy of the |NCS| source code, which will be
+self-contained in a single folder, you can then proceed to manage that code in
+any way you see fit.
+
+Worflow 2: Out of tree application repository
+---------------------------------------------
+
+A very simple approach to maintaining your application is to completely decouple
+it from the |NCS| repositories and instead host it wherever you prefer (be it
+Git, another version control system or simply on your hard drive). This is
+typically also known as "out of tree" application, meaning that the application,
+boards definitions and any other libraries are actually outside any of the
+repositories provided by the |NCS| and can in fact be placed anywhere at all.
+As long as you do not need to make any changes to any of the repositories of the
+|NCS|, you can use the procedures to :ref:`get the source code <dm-wf-get-ncs>`
+and later :ref:`update it <dm-wf-update-ncs>`, and manage your application
+separately, inside or outside the top folder of the |NCS|.
+
+If you choose to have your application outside the folder hierarchy of the
+|NCS|, the build system will find the location of the SDK through the
+:makevar:`ZEPHYR_BASE` environment variable, which is set either through a
+script or manually in an IDE. More information about application development and
+the |NCS| build and configuration system can be found in the :ref:`ncs-app-dev`
+documentation section.
+
+Worflow 3: App in a fork of `fw-nrfconnect-nrf`_
+------------------------------------------------
+
+Forking the `fw-nrfconnect-nrf`_ repository and adding the application to it is
+another valid option to develop and maintain your application. In order to use
+this approach, you will first need to :ref:`get the source code
+<dm-wf-get-ncs>`, and then :ref:`fork the fw-nrfconnect-nrf repository
+<dm-wf-fork>`. Once you have your own fork you can start adding your application
+to your fork's tree and push it to your own Git server. Every time you want to
+update the revision of the |NCS| that you want to use as a basis for your
+application you will need to follow the :ref:`instructions to update
+<dm-wf-update-ncs>` on your own fork of ``fw-nrfconnect-nrf``.
+
+Workflow 4: App as the manifest repository
```

Comment:
This is my second favorite workflow compared to workflow 3. The main benefit compared to 3 is it allows full control of the manifest file itself, which might be worth highlighting. The main drawback is the same :smile: .